code/modules/mining/drilling/drill.dm

```diff
@@ -131,12 +135,19 @@
 		active = 0
 		need_player_check = 1
 		update_icon()
+		system_error("resources depleted")
 
 /obj/machinery/mining/drill/attack_ai(var/mob/user as mob)
 	return src.attack_hand(user)
 
 /obj/machinery/mining/drill/attackby(obj/item/O as obj, mob/user as mob)
 	if(!active)
+		if(istype(O, /obj/item/device/multitool))
+			var/newtag = text2num(sanitizeSafe(input(user, "Enter new ID number or leave empty to cancel.", "Assign ID number") as text, 4))
+			if(newtag)
+				name = "[initial(name)] #[newtag]"
+				user << "<span class='notice'>You changed the drill ID to: [newtag]</span>"
```

Comment:
to_chat(user, "<span class='notice'>You changed the drill ID to: [newtag]</span>")